test/fixtures/browser-app/app.js

```diff
@@ -7,4 +7,6 @@ var loopback = require('loopback');
 var boot = require('../../../');
 
 var app = module.exports = loopback();
-boot(app);
+app.start = function(done) {
+  boot(app, __dirname, done);
+};
```

Comment:
> When the app is used as a dependency or library, it will be much more flexible to defer the boot process.

I don't understand. If you defer the boot process, then you are getting an empty loopback application created by `var app = loopback()`. What is such an empty app good for?

Even if we agree that it's a good idea to defer the booting process, then we should use a different name than `start` because that one is already used in our project template for something else, see https://github.com/strongloop/loopback-workspace/blob/43bafdba46eed1ee3f4152951c847e299e967b34/templates/projects/empty-server/files/server/server.js#L8 I am proposing to use `app.setup`, `app.init` or `app.boot` instead. (Note that `app.boot` clashes with LoopBack 1.x `app.boot`, but I don't think that's a big deal.)